### PR TITLE
Baremetal boot, VGA driver, double-fault handler

### PR DESCRIPTION
**Pull request by phanikishoreg**

### CODE REVIEW

File: src/platform/i386/kernel.c [hunk 1]
```diff
@@ -67,14 +69,19 @@ kern_memory_setup(struct multiboot *mb, u32_t mboot_magic)
 	for (i = 0 ; i < mb->mods_count ; i++) {
 		struct multiboot_mod_list *mod = &mods[i];
```
Comment (gparmer) [inline src/platform/i386/kernel.c]: Remove blank lines between definitions.

File: src/platform/i386/kernel.c [hunk 2]
```diff
@@ -67,14 +69,19 @@ kern_memory_setup(struct multiboot *mb, u32_t mboot_magic)
...
 
-		printk("\t- %d: [%08x, %08x)", i, mod->mod_start, mod->mod_end);
+		char *cmdline = (char *)mod->cmdline;
+		int cmdline_len = strnlen((const char*)cmdline, CMDLINE_MAX_LEN);
+		assert(cmdline_len >= CMDLINE_REQ_LEN);
```
Comment (gparmer) [inline src/platform/i386/kernel.c]: Assertion should be _after_ the definitions.  (don't mix definitions and statements)

File: src/kernel/include/thd.h [hunk 3]
```diff
@@ -92,7 +92,7 @@ thd_upcall_setup(struct thread *thd, u32_t entry_addr, int option, int arg1, int
 {
 	struct pt_regs *r = &thd->regs;
 
-	thd->state |= THD_STATE_PREEMPTED;
+	/*thd->state |= THD_STATE_PREEMPTED;*/
```
Comment (gparmer) [inline src/kernel/include/thd.h]: Please get rid of this.

File: src/platform/i386/vga.c [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+
+/* Clears the screen */
+static void cls(void)
+{
+	unsigned blank = 0;
+	int i = 0;
+
+	/* Again, we need the 'short' that will be used to
+	 *  represent a space with color */
```
Comment (gparmer) [inline src/platform/i386/vga.c]: Multi-line comments should be like:
```
/*
 *
 */
```

Please read the style guide

File: src/platform/i386/vga.c [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+	blank = SPACE | (attrib << 8);
+
+	/* Sets the entire screen to spaces in our current
+	 *  color */
+	for(i = 0; i < LINES; i++)
```
Comment (gparmer) [inline src/platform/i386/vga.c]: Always use `{}` unless the body is on the same line as the conditional.  Style guide.

File: src/platform/i386/vga.c [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+
+/* Put the character C on the screen. */
+static void putchar (int c)
+{
+	if (c == '\n' || c == '\r') {
+newline:
+		cll();
+		csr_x = 0;
+		csr_y++;
+		if (csr_y >= LINES)
```
Comment (gparmer) [inline src/platform/i386/vga.c]: Same.

File: src/platform/i386/vga.c [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+			csr_y = 0;
+		return;
+	}
+
+	*(video + (csr_x + csr_y * COLUMNS) * 2) = c & 0xFF;
+	*(video + (csr_x + csr_y * COLUMNS) * 2 + 1) = attrib;
+
+	csr_x++;
+	if (csr_x >= COLUMNS)
+		goto newline;
```
Comment (gparmer) [inline src/platform/i386/vga.c]: same.  This is a good example where I might do 
`if (csr_x >= COLUMNS) goto newline;`
instead.

File: src/platform/i386/vga.c [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+}
+
+void vga_puts(const char *s)
+{
```
Comment (gparmer) [inline src/platform/i386/vga.c]: Single-line functions can be 
```
type
fnname(params)
{ body }
```

You *have* to put the type on a line on its own.  You should put the body on the same line as the `{}` if it is a single line.

File: src/platform/i386/vga.c [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+	puts((unsigned char *)s);
+}
+
+/* Uses the above routine to output a string... */
+static void puts(unsigned char *text)
```
Comment (gparmer) [inline src/platform/i386/vga.c]: Fix this.  the type should be on a seprate line for each of the fns.

File: src/platform/i386/vm.c [hunk 10]
```diff
@@ -19,6 +19,12 @@ u32_t boot_comp_pgd[PAGE_SIZE/sizeof(u32_t)] PAGE_ALIGNED = {
 };
 
 void
+double_fault_handler(struct pt_regs *regs)
+{
```
Comment (gparmer) [inline src/platform/i386/vm.c]: same rule as above for single line functions.

File: src/platform/i386/kernel.c (hunk 1 above)
Comment (gparmer) [inline src/platform/i386/kernel.c]: Is there a blank line after this definition?  We don't want blank lines between different defintions.

File: src/platform/i386/kernel.c (hunk 2 above)
Comment (phanikishoreg) [inline src/platform/i386/kernel.c]: thank you. i didn't notice that. sorry..

File: src/kernel/include/thd.h (hunk 3 above)
Comment (phanikishoreg) [inline src/kernel/include/thd.h]: i will do it.

File: src/platform/i386/vga.c (hunk 4 above)
Comment (phanikishoreg) [inline src/platform/i386/vga.c]: sorry.. i'll make those changes.

File: src/platform/i386/vga.c (hunk 5 above)
Comment (phanikishoreg) [inline src/platform/i386/vga.c]: quickly read through the style guide. will make necessary changes. thank you.

File: src/platform/i386/kernel.c (hunk 1 above)
Comment (phanikishoreg) [inline src/platform/i386/kernel.c]: will fix it too.

**Comment by phanikishoreg** [comment]
sorry and thank you for your review comments. i'm closing this request. i'll redo the commits and pull request.

**Pull request closed without merging**